globus_sdk/transfer/paging.py

```diff
@@ -163,22 +179,68 @@ def iterable_func(self):
 
         This method is the real workhorse of this entire module.
         """
-        # now, cap the limit per request to the max per request size
-        limit = min(self.num_results, self.max_results_per_call)
+        if not self.client_kwargs['params']:
+            self.client_kwargs['params'] = {}
+
+        # to start with, cap the limit per request to the max per request size
+        self.limit = min(self.num_results, self.max_results_per_call)
+
+        def _set_params_for_next_call():
+            # if we're about to request more results than the user asked
+            # for, limit ourselves on the last paginated call to the API
+            if self.offset + self.limit > self.num_results:
+                self.limit = self.num_results - self.offset
+
+            # if the paging is done by marker, just carry over the marker
+            if self.paging_style == self.PAGING_STYLE_MARKER:
+                if self.marker:
+                    self.client_kwargs['params']['marker'] = self.marker
+            # these params work for all paging styles *except* MARKER
+            else:
+                self.client_kwargs['params']['offset'] = self.offset
+                self.client_kwargs['params']['limit'] = self.limit
+
+        def _check_has_next_page(res):
+            """
+            Check that the API says there are more results available.
+
+            Additionally, update the PaginatedResource.maker or
+            PaginatedResource.offset based on the response
+            """
+            # if the paging style is MARKER, look at the marker
+            if self.paging_style == self.PAGING_STYLE_MARKER:
+                # marker may be 0, null, or absent if no more results
+                # API docs aren't 100% clear -- looks like 0 is what we should
+                # expect, but we'll also accept null or absent to be safe
```

Comment:
You are 100% correct. And yet, I remember this working... Perhaps I got confused in long lists of files and just saw the same results over and over. Manual validation at its finest.

I should be fetchng `next_marker` -- for this pattern of purely paging results, there's no reason to keep around the value of `marker`. I'll fix and re-test, more carefully this time.